### DISCUSSION TITLE
Обновление приложения на F-Droid

### DISCUSSION THREAD

**Discussion by Komprogue** [RHVoice/RHVoice#432] [Q&A]
Будет ли обновлен билд на f-droid? Там до сих пор лежит старая [версия 1.2.4](https://f-droid.org/ru/packages/com.github.olga_yakovleva.rhvoice.android/), хотя уже прошло четыре недели с момента релиза новой версии.

**Comment by alex19EP** [comment]
я думаю этот вопрос надо задать fdroid.
насколько я помню - мы сделали всё что они просили для автоматического обновления.

**Reply by TheLastProject** [reply]
Hi, F-Droid maintainer here. This question was asked on the forum last time but apparently the answer wasn't communicated back to you.

F-Droid can't autoupdate because your build.gradle calculates the versionCode dynamically which isn't supported during F-Droid update checking: https://forum.f-droid.org/t/rhvoice-updating/16441/3?u=thelastproject

It would be awesome if you could look into one of those options to make it possible for F-Droid to autoupdate in the future :)

**Comment by alex19EP** [comment]
ok let's see what @Olga-Yakovleva will say.

I can definitely add a txt file when creating a release, but I didn’t quite understand what exactly needs to be written there.

**Reply by TheLastProject** [reply]
For the text file method, Wire puts the following in theirs (I changed it to your values for 1.8.0):

```
VersionCode: 10800
VersionName: 1.8.0
```

So attaching this as version.txt or fdroid.txt or whatever you want (as long as the filename stays the same each release) would work.

**Comment by Olga-Yakovleva** [comment]
Let's use this version.txt method. Just in case, a correction for the 
current release:

VersionCode: 108000

**Comment by alex19EP** [comment]
ok @TheLastProject I uploaded fdroid.txt to 1.8.0 release. will keep that in sync.

**Reply by TheLastProject** [reply]
Thanks! I have set up F-Droid's update checking to read the fdroid.txt of whatever your latest release is and grab the version info from there. So RHVoice updates will now be detected by F-Droid :)

https://gitlab.com/fdroid/fdroiddata/-/merge_requests/10900
